spec/unit/element_filter_spec.rb

```diff
@@ -31,136 +31,231 @@
+          to be(true)
       end
 
-      it "does not match on extra elements" do
-        expect(@filter.matches?(Machinery::Array.new(["a", "b", "c"]))).to be(false)
+      it "returns true when one matcher doesn't match" do
+        filter = ElementFilter.new(@path, Filter::OPERATOR_EQUALS_NOT, [@matcher2, @matcher1])
+        expect(filter.matches?("/home/alfred")).
+          to be(true)
       end
 
-      it "does not match on missing elements" do
-        expect(@filter.matches?(Machinery::Array.new(["a"]))).to be(false)
+      describe "matches beginning of a value" do
+        before(:each) do
+          @filter = ElementFilter.new("path", Filter::OPERATOR_EQUALS_NOT, "/home/alfred/*")
+        end
+
+        it "returns true on shorter value" do
+          expect(@filter.matches?("/home/alfred")).to be(true)
+        end
+
+        it "returns false on minimal match" do
+          expect(@filter.matches?("/home/alfred/")).to be(false)
+        end
+
+        it "returns false on longer match" do
+          expect(@filter.matches?("/home/alfred/and/berta")).to be(false)
+        end
+
+        it "returns false on value with star at the end" do
+          expect(@filter.matches?("/home/alfred/*")).to be(false)
+        end
       end
 
-      it "allows for filtering arrays with one element using a string filter" do
-        filter = ElementFilter.new("path", ["a"])
-        expect(filter.matches?(Machinery::Array.new(["a"]))).to be(true)
-        expect(filter.matches?(Machinery::Array.new(["a", "b"]))).to be(false)
+      context "matching arrays" do
+        before(:each) do
+          @filter = ElementFilter.new("path", Filter::OPERATOR_EQUALS_NOT, [["a", "b"]])
+        end
+
+        it "does not match equal arrays" do
+          expect(@filter.matches?(Machinery::Array.new(["a", "b"]))).to be(false)
+        end
+
+        it "matches on extra elements" do
+          expect(@filter.matches?(Machinery::Array.new(["a", "b", "c"]))).to be(true)
+        end
+
+        it "matches on missing elements" do
+          expect(@filter.matches?(Machinery::Array.new(["a"]))).to be(true)
+        end
+
+        it "allows for filtering arrays with one element using a string filter" do
+          filter = ElementFilter.new("path", Filter::OPERATOR_EQUALS_NOT, ["a"])
+          expect(filter.matches?(Machinery::Array.new(["a"]))).to be(false)
+          expect(filter.matches?(Machinery::Array.new(["a", "b"]))).to be(true)
+        end
       end
     end
   end
 
   describe "filters_scope?" do
-    specify { expect(ElementFilter.new("/foo", "bar").filters_scope?("foo")).to be(true) }
-    specify { expect(ElementFilter.new("/foo", "bar").filters_scope?("foo_bar")).to be(false) }
+    specify { expect(ElementFilter.new("/foo", Filter::OPERATOR_EQUALS_NOT, "bar").filters_scope?("foo")).to be(true) }
+    specify { expect(ElementFilter.new("/foo", Filter::OPERATOR_EQUALS_NOT, "bar").filters_scope?("foo_bar")).
```

Comment:
Line is too long. [110/100]<br>Block body expression is on the same line as the block start.